### PR TITLE
Update test of fracture propagation

### PR DESCRIPTION
**Pull request by keileg**
## Proposed changes
Use pytest.autouse on a few setup methods. Change the name of one class to make it visible to pytest.

## Types of changes

What types of changes does this PR introduce to PorePy?
_Put an `x` in the boxes that apply._

- [ ] Minor change (e.g., dependency bumps, broken links).
- [ ] Bugfix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [x] Testing (contribution related to testing of existing or new functionality).
- [ ] Documentation (contribution related to adding, improving, or fixing documentation).
- [ ] Maintenance (e.g., improve logic and performance, remove obsolete code).
- [ ] Other:

## Checklist

_Put an `x` in the boxes that apply or explain briefly why the box is not relevant._

- [x] The documentation is up-to-date.
- [x] Static typing is included in the update.
- [x] This PR does not duplicate existing functionality.
- [x] The update is covered by the test suite (including tests added in the PR).
- [ ] If new skipped tests have been introduced in this PR, `pytest` was run with the `--run-skipped` flag.

### CODE REVIEW

File: tests/numerics/fracture_deformation/test_fracture_propagation.py [hunk 1]
```diff
@@ -582,7 +582,7 @@ class PropagationCriteria:
         values for u and parameters.
 
     """
-
+    @pytest.fixture(autouse=True)
```
Comment (IvarStefansson) [inline tests/numerics/fracture_deformation/test_fracture_propagation.py:585]: Should we be explicit about the scope of the fixture?

Comment (keileg) [reply tests/numerics/fracture_deformation/test_fracture_propagation.py:585]: I don't use it that way, but I don't know if I follow best practice.

Comment (IvarStefansson) [reply tests/numerics/fracture_deformation/test_fracture_propagation.py:585]: Let me rephrase: Have you considered what the scope of this fixture is? 
I think that it is `function`, i.e., it is invoked once per `test_...` method of this class. Do you agree? And is this the intended behaviour?

Comment (keileg) [reply tests/numerics/fracture_deformation/test_fracture_propagation.py:585]: No, I have not considered this.

This is a fix of functionality which in practice is not used, so honestly, I put a low priority on maintaining the tests beyond making sure they run, and do take too much time (this class runs pretty fast).
